### PR TITLE
Use an Annotation to mark graph ql queries.

### PR DESCRIPTION
**Pull request by simonseyock**
This introduces an annotation `@GraphQLQuery` that annotates a method that returns a `DataFetcher` for a graph ql query. The GraphQLProvider will search for all methods with this annotation automatically on application start up.

Usage:
```
    @GraphQLQuery(name = "allUsers")
    public DataFetcher getAllUsers() {
        return dataFetchingEnvironment -> this.userService.findAll();
    }
```

It can be used without the name parameter in which case the method name is used.

It has to be checked that the GraphQLProvider will correctly search for the annotation if used from a child project.

### CODE REVIEW

File: shogun-lib/src/main/resources/graphql/shogun.graphqls [hunk 1]
```diff
@@ -11,6 +11,7 @@ type Query {
     fileById(id: Int): File
     fileByUuid(uuid: String): File
     allFiles: [File]
+    getImageFileByUuid(uuid: String): File
```
Comment (KaiVolland) [inline shogun-lib/src/main/resources/graphql/shogun.graphqls:14]: Please remove the `get` to be consistent.

**Comment by KaiVolland** [comment]
Very nice!

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/graphql/GraphQLDataFetchers.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public DataFetcher getAllFiles() {
     }
 
     // ImageFile
-    @GraphQLQuery(name = "getImageFileByUuid")
+    @GraphQLQuery(name = "imageFileByUuid")
```
Comment (KaiVolland) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/graphql/GraphQLDataFetchers.java:88]: You missed to add the change to the schema file. :)

Comment (simonseyock) [reply shogun-lib/src/main/java/de/terrestris/shogun/lib/graphql/GraphQLDataFetchers.java:88]: true

File: shogun-lib/src/main/resources/graphql/shogun.graphqls (hunk 1 above)
Comment (simonseyock) [inline shogun-lib/src/main/resources/graphql/shogun.graphqls:14]:
```suggestion
    imageFileByUuid(uuid: String): File
```

**Review by KaiVolland — APPROVED** [review]
Thanks. Please merge.

**Pull request merged**